hack/verify-archives.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# About:
+# This script extracts the golden image tarballs and compares their contents against the
+# already-tracked golden folder (which is used by Bazel to generate deterministic images
+# on every e2e test invocation). The purpose of this script is to verify that the tarballs
+# residing in golden-archives matches the original source of truth (golden).
+#
+# Usage:
+#   verify-archives.sh repo-root
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [[ "$#" != 1 ]]
+then
+    echo "Error: required argument missing!" >&2
+    echo "Usage verify-archives.sh repo-root" >&2
+    exit 1
+fi
+
+repoRoot=$1
+# Mapping for each golden image KEY=localPath VALUE=containerPath.
+declare -A paths
+paths[bar/1.0]=bar/1.0
+paths[foo/1.0-linux_amd64]=foo/linux_amd64/1.0-linux_amd64
+paths[foo/1.0-linux_s390x]=foo/linux_s390x/1.0-linux_s390x
+paths[foo/NOTAG-0]=foo/NOTAG/NOTAG-0
+
+# Setup mock directory structure.
+mockGolden=$(mktemp -d)
+mkdir "${mockGolden}/foo"
+mkdir "${mockGolden}/bar"
+
+goldenArchive="${repoRoot}/test-e2e/cip/golden-archives/"
+goldenContent="${repoRoot}/test-e2e/cip/golden"
+
+# Looping through localPath archives.
+for path in "${!paths[@]}"
+do
+    containerPath=${paths[$path]}
+    # Get parent directory of file.
+    goldenParent="$(cut -d "/" -f1 <<< "$path")"
```

Comment:
Would `goldenParent=${path%%/*}` be the equivalent? To quote from the Bash manual:

```
       ${parameter#word}
       ${parameter##word}
              Remove  matching  prefix pattern.  The word is expanded to produce a pattern just as in pathname expansion, and matched against
              the expanded value of parameter using the rules described under Pattern Matching below.  If the pattern matches  the  beginning
              of  the value of parameter, then the result of the expansion is the expanded value of parameter with the shortest matching pat‐
              tern (the ``#'' case) or the longest matching pattern (the ``##'' case) deleted.  If parameter is @ or *, the  pattern  removal
              operation  is  applied to each positional parameter in turn, and the expansion is the resultant list.  If parameter is an array
              variable subscripted with @ or *, the pattern removal operation is applied to each member of the array in turn, and the  expan‐
              sion is the resultant list.

       ${parameter%word}
       ${parameter%%word}
              Remove  matching  suffix pattern.  The word is expanded to produce a pattern just as in pathname expansion, and matched against
              the expanded value of parameter using the rules described under Pattern Matching below.  If the pattern matches a trailing por‐
              tion  of the expanded value of parameter, then the result of the expansion is the expanded value of parameter with the shortest
              matching pattern (the ``%'' case) or the longest matching pattern (the ``%%'' case) deleted.  If parameter is @ or *, the  pat‐
              tern  removal operation is applied to each positional parameter in turn, and the expansion is the resultant list.  If parameter
              is an array variable subscripted with @ or *, the pattern removal operation is applied to each member of the array in turn, and
              the expansion is the resultant list.
```

Reply:
ack.